python/tk_desktop/rpc.py

```diff
@@ -219,41 +336,49 @@ def __init__(self, pipe, authkey):
             family = "AF_PIPE"
         else:
             family = "AF_UNIX"
-        logger.debug("client connecting to to %s", pipe)
+        logger.debug("multi client connecting to to %s", pipe)
         self._connection = multiprocessing.connection.Client(
             address=pipe, family=family, authkey=authkey
         )
-        logger.debug("client connected to %s", pipe)
+        logger.debug("multi client connected to %s", pipe)
 
     def call_no_response(self, name, *args, **kwargs):
-        msg = "client calling '%s(%s, %s)'" % (name, args, kwargs)
+        msg = "multi client calling '%s(%s, %s)'" % (name, args, kwargs)
         if self._closed:
-            raise EOFError("closed " + msg)
+            raise RuntimeError("closed " + msg)
         # send the call through with args and kwargs
         logger.debug(msg)
         self._connection.send(pickle.dumps((False, name, args, kwargs)))
 
     def call(self, name, *args, **kwargs):
-        msg = "client waiting call '%s(%s, %s)'" % (name, args, kwargs)
+        msg = "multi client waiting call '%s(%s, %s)'" % (name, args, kwargs)
         if self._closed:
-            raise EOFError("closed " + msg)
+            raise RuntimeError("closed " + msg)
         # send the call through with args and kwargs
         logger.debug(msg)
         self._connection.send(pickle.dumps((True, name, args, kwargs)))
 
         # wait until there is a result, pause to check if we have been closed
         while True:
-            if self._connection.poll(self.LISTEN_TIMEOUT):
-                # have a response waiting, grab it
-                break
-            else:
-                # no response waiting, see if we need to stop the client
+            try:
+                if self._connection.poll(self.LISTEN_TIMEOUT):
+                    # have a response waiting, grab it
+                    break
+                else:
+                    # If the connection was closed while we polled, raise an error.
+                    if self._closed:
+                        raise RuntimeError("client closed while waiting for a response")
```

Comment:
I noticed that you altered a previous message to become `multi client ...` should this also be updated as well as the raise a few lines below?

Reply:
This message is protocol agnostic and the same for both protocols, so I'd leave it as is.